File: 2_gyak/Liskov/C++/Liskov.cpp
#include <iostream>

using namespace std;

class Madarak {
		
public:
	void repul(){
		cout << "repulok ..." << endl;
	}

};

class Pingvin: public Madarak {

};

class Sas : public Madarak {

};

int main ( int argc, char **argv ){
     
	Pingvin pingvin;
	Sas sas;

	pingvin.repul();
	sas.repul();

	return 0;
}

File: 2_gyak/Liskov/Java/Liskov.java
public class Liskov {

	class Madarak{

		
		void repul(){
			System.out.println( this.toString() + " ... repul ... ");
		}

	}

	class Pingvin extends Madarak{

	}
	
	class Sas extends Madarak{

	}
	
	public static void main(String[] args) {
		Liskov liskov = new Liskov();
		
		Pingvin pingvin = liskov.new Pingvin();
		Sas sas = liskov.new Sas();
		
		pingvin.repul();
		sas.repul();
	}
}